pywr-schema/src/timeseries/mod.rs

```diff
@@ -0,0 +1,153 @@
+mod polars_dataset;
+
+use ndarray::{Array1, Array2};
+use polars::error::PolarsError;
+use polars::prelude::DataType::Float64;
+use polars::prelude::{DataFrame, Float64Type, IndexOrder, Schema};
+use pywr_core::models::ModelDomain;
+use pywr_core::parameters::{Array1Parameter, Array2Parameter, ParameterIndex};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+use thiserror::Error;
+
+use crate::{parameters::ParameterMeta, SchemaError};
+
+use self::polars_dataset::PolarsDataset;
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+#[serde(tag = "type")]
+enum TimeseriesProvider {
+    Pandas,
+    Polars(PolarsDataset),
+}
+
+struct ResamplingArgs {}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+pub struct Timeseries {
+    #[serde(flatten)]
+    names: ParameterMeta,
+    provider: TimeseriesProvider,
+    url: PathBuf,
```

Comment:
I think we discussed this might be defined in each provider as required. There might be providers that do not need a URL.